### PR TITLE
Remove dom4 as a dependency in core

### PR DESCRIPTION
**Pull request by rcchen**
#### Changes proposed in this pull request:

Removes `dom4` as a dependency. Should theoretically unblock SSR per #131 though that use case is currently untested with this PR

#### Reviewers should focus on:

* Does this approach look sane?
* Do things continue to work as they did before?

### CODE REVIEW

File: packages/core/src/common/utils.ts [hunk 1]
```diff
@@ -50,6 +50,33 @@ export function clamp(val: number, min: number, max: number) {
     return Math.min(Math.max(val, min), max);
 }
 
+/* Alternate implmentation of Element.closest */
+export function closest(element: Node, selector: string) {
```
Comment (crazytoucan) [inline packages/core/src/common/utils.ts]: Why isn't the native one good enough? If this is for polyfill/compatibility goodness, is there any way to fall back to the native?

**Comment by rcchen** [comment]
Looks like there are things like `KeyboardEvent` that are having problems with Phantom; thoughts on keeping `dom4` for testing or figuring out how to solve this in Phantom?

**Comment by codebling** [comment]
There are a few changes beyond the removal of that need to happen for server rendering to work. Checkout [my fork](https://github.com/codebling/blueprint/tree/server-friendly). 

I don't think that dom4 needs to be removed entirely, as long as it doesn't run on the server. In my fork, we [conditionally load dom4](https://github.com/codebling/blueprint/commit/e9c89e06c3823db1aef1580bc47d3d518a12cdef), which allows server-side rendering while requiring only minimal code changes and permits the project members to continue using the same workflow. 

I'm sorry I didn't post [this](https://github.com/palantir/blueprint/issues/131#issuecomment-266630449) earlier, we could have saved some duplication of work.

**Comment by adidahiya** [comment]
@rcchen you'll need to enable [Circle CI for your fork](https://circleci.com/gh/rchen/blueprint) for status to be reported on this PR

**Comment by codebling** [comment]
See #360

**Comment by adidahiya** [comment]
I prefer @codebling's approach to this problem. Despite dom4 being _a little_ overkill (it makes browser behavior consistent all the way back to IE8, while we only support IE10+), it is nice to have and I particularly like a couple features, like making the interface for `queryAll` results look like `Array<Element>`. And I would feel safer with a smaller code change in the DOM manipulation code across the board. We may decide to reëvaluate dom4 later, but this seems like a good approach to supporting SSR at the moment.

**Comment by rcchen** [comment]
Cool, let's proceed with the other and revisit `dom4` at another time. I'll close out this PR for now 🎉

**Pull request closed without merging**